middleware/security_context_setter.rb

```diff
@@ -1,16 +1,29 @@
 module CloudFoundry
   module Middleware
     class SecurityContextSetter
+      PACKAGES_UPLOAD_REGEX = %r{/v3/packages/[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}/upload}i.freeze
+
       def initialize(app, security_context_configurer)
         @app                         = app
         @security_context_configurer = security_context_configurer
       end
 
       def call(env)
         header_token = env['HTTP_AUTHORIZATION']
+        request_path = env['REQUEST_PATH']
 
         @security_context_configurer.configure(header_token)
```

Comment:
Is my understanding correct that this line will set an "invalid_token" security context and below it will be overwritten in case the upload start time is provided? If so it would be better to put this into an `else` block. Or do something like this:
```
security_context_configurer = @security_context_configurer

if ...
  security_context_configurer = VCAP::CloudController::Security::SecurityContextConfigurer.new(relaxed_token_decoder)
end

security_context_configurer.configure(header_token)
```

Reply:
Discussed offline, and I've resolved this the way you suggest.